src/main/java/com/google/cloud/spanner/pgadapter/wireprotocol/WireMessage.java

```diff
@@ -186,6 +186,11 @@ protected int getHeaderLength() {
    * setting for {@link ConnectionHandler}.
    */
   public void nextHandler() throws Exception {
-    this.connection.setMessageState(ControlMessage.create(this.connection));
+    try {
+      this.connection.setMessageState(ControlMessage.create(this.connection));
+    } catch (Exception e) {
+      System.out.println("**** " + e.toString());
+      throw e;
+    }
```

Comment:
Is this intentional? I would expect these exceptions to be handled at a higher level....

Reply:
Not intentional. Removed now.